sonic_platform_base/sensor_base.py

```diff
@@ -0,0 +1,172 @@
+        Retrieves measurement reported by sensor
+
+        Returns:
+            Sensor measurement
+        """
+        raise NotImplementedError
+
+    def get_unit(self):
+        """
+        Retrieves unit of measurement reported by sensor
+
+        Returns:
+            Sensor measurement unit
+        """
+        raise NotImplementedError
+
+    def get_high_threshold(self):
+        """
+        Retrieves the high threshold of sensor
+
+        Returns:
+            High threshold 
+        """
+        raise NotImplementedError
+
+    def get_low_threshold(self):
+        """
+        Retrieves the low threshold 
+
+        Returns:
+            Low threshold 
+        """
+        raise NotImplementedError
+
+    def set_high_threshold(self, value):
+        """
+        Sets the high threshold value of sensor
+
+        Args :
+            value: High threshold value to set
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def set_low_threshold(self, value):
+        """
+        Sets the low threshold value of sensor
+
+        Args :
+            value: Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def get_high_critical_threshold(self):
+        """
+        Retrieves the high critical threshold value of sensor
+
+        Returns:
+            The high critical threshold value of sensor 
+        """
+        raise NotImplementedError
+
+    def get_low_critical_threshold(self):
+        """
+        Retrieves the low critical threshold value of sensor
+
+        Returns:
+            The low critical threshold value of sensor 
+        """
+        raise NotImplementedError
+
+    def set_high_critical_threshold(self, value):
+        """
+        Sets the critical high threshold value of sensor
+
+        Args :
+            value: Critical high threshold Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def set_low_critical_threshold(self, value):
+        """
+        Sets the critical low threshold value of sensor
+
+        Args :
+            value: Critial low threshold Value 
+
+        Returns:
+            A boolean, True if threshold is set successfully, False if not
+        """
+        raise NotImplementedError
+
+    def get_minimum_recorded(self):
+        """
+        Retrieves the minimum recorded value of sensor
+
+        Returns:
+            The minimum recorded value of sensor 
+        """
+        raise NotImplementedError
+
+    def get_maximum_recorded(self):
+        """
+        Retrieves the maximum recorded value of sensor
+
+        Returns:
+            The maximum recorded value of sensor 
+        """
+        raise NotImplementedError
+
+
+
+class VsensorBase(SensorBase):
```

Comment:
Same question about naming for the classes, more explicit would be beneficial I think. Additionally, should these be in separate files? I don't remember off the top of my head any existing examples in SONiC of multiple base classes in the same file.

Reply:
In general i think we should have a separate files. However I think it makes sense here to provide an easy reference.
Will check if there is strict coding guideline for this.